exonum-java-binding-cryptocurrency-demo/src/main/java/com/exonum/binding/cryptocurrency/transactions/TransferTx.java

```diff
@@ -66,36 +66,27 @@
   /**
    * Creates a new transfer transaction from the binary message.
    */
-  public static TransferTx fromMessage(BinaryMessage message) {
-    checkTransaction(message, ID);
+  public static TransferTx fromRawTransaction(RawTransaction rawTransaction) {
+    checkTransaction(rawTransaction, ID);
 
-    try {
-      TxMessageProtos.TransferTx messageBody =
-          TxMessageProtos.TransferTx.parseFrom(message.getBody());
+    TxMessageProtos.TransferTx body =
+        PROTO_SERIALIZER.fromBytes(rawTransaction.getPayload());
 
-      long seed = messageBody.getSeed();
-      PublicKey fromWallet = toPublicKey(messageBody.getFromWallet());
-      PublicKey toWallet = toPublicKey(messageBody.getToWallet());
-      long sum = messageBody.getSum();
+    long seed = body.getSeed();
+    PublicKey fromWallet = toPublicKey(body.getFromWallet());
+    PublicKey toWallet = toPublicKey(body.getToWallet());
+    long sum = body.getSum();
 
-      return new TransferTx(message, seed, fromWallet, toWallet, sum);
-    } catch (InvalidProtocolBufferException e) {
-      throw new IllegalArgumentException("Invalid TxMessageProtos.TransferTx buffer", e);
-    }
+    return new TransferTx(seed, fromWallet, toWallet, sum);
   }
 
   private static PublicKey toPublicKey(ByteString s) {
     return PublicKey.fromBytes(s.toByteArray());
   }
 
   @Override
-  public boolean isValid() {
-    return getMessage().verify(CRYPTO_FUNCTION, fromWallet);
-  }
-
-  @Override
-  public void execute(Fork view) throws TransactionExecutionException {
-    CryptocurrencySchema schema = new CryptocurrencySchema(view);
+  public void execute(TransactionContext context) throws TransactionExecutionException {
```

Comment:
I think with manual verification of the signature against `from` gone,
    we **must** check that authorPk == fromWallet, or, better, remove `fromWallet` and use `authorPk` as `fromWallet` (because no one but `fromWallet` can sign this tx).